### PR TITLE
Remove legacy functions

### PR DESCRIPTION
**Pull request by jacobrgardner**
This PR converts all GPyTorch Functions to the non legacy format, and removes `NormalCDF`, which torch now has native support for via `torch.distributions.Normal(0., 1.).cdf(...)`

### CODE REVIEW

File: gpytorch/functions/_log_normal_cdf.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import math
-from . import NormalCDF
 from torch.autograd import Function
 import torch
+from torch.distributions import Normal
 
 
 class LogNormalCDF(Function):
+    @staticmethod
     def forward(self, z):
```
Comment (Balandat) [inline gpytorch/functions/_log_normal_cdf.py]: `ctx` to follow consistent convention?

File: gpytorch/functions/_log_normal_cdf.py [hunk 2]
```diff
@@ -90,11 +91,12 @@ def forward(self, z):
...
             self.numerator = numerator
 
-        log_phi_z.masked_scatter_(z_is_ordinary, torch.log(NormalCDF().forward(z.masked_select(z_is_ordinary))))
+        log_phi_z.masked_scatter_(z_is_ordinary, torch.log(Normal(0., 1.).cdf(z.masked_select(z_is_ordinary))))
 
         self.save_for_backward(z, log_phi_z)
         return log_phi_z
 
+    @staticmethod
     def backward(self, grad_output):
```
Comment (Balandat) [inline gpytorch/functions/_log_normal_cdf.py]: `ctx`

Comment (jacobrgardner) [reply gpytorch/functions/_log_normal_cdf.py]: Good catch, updated.

**Comment by Balandat** [comment]
thanks for the quick fix!

**Pull request merged**